spock_apply.c

```diff
@@ -2752,6 +2869,24 @@ apply_work(PGconn *streamConn)
 					if (last_received < end_lsn)
 						last_received = end_lsn;
 
+					w_message_count++;
+
+					/*
+					 * Send feedback if wal_sender_timeout/2 has passed or after 10 'w' messages.
+					 */
+					if (TimestampDifferenceExceeds(last_receive_timestamp, GetCurrentTimestamp(), wal_sender_timeout / 2) ||
```

Comment:
GetCurrentTimestamp() is being called repeatedly in this code block. We should only call it once and save it in a local variable.